### PR TITLE
fix(ingest/unity): creating group urn in case of group

### PR DESCRIPTION
**Pull request by dushayntAW**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/unity/proxy.py [hunk 1]
```diff
@@ -187,6 +187,12 @@ def service_principals(self) -> Iterable[ServicePrincipal]:
             if optional_sp:
                 yield optional_sp
 
+    def groups(self):
+        group_list: List[str] = []
+        for group in self._workspace_client.groups.list():
```
[the remark this answers is not in the document]
Comment (dushayntAW) [reply metadata-ingestion/src/datahub/ingestion/source/unity/proxy.py:192]: @mayurinehate no, groups we are fetching directly using the workspace client that we have already created and doesn't require any additional permission

File: metadata-ingestion/src/datahub/ingestion/source/unity/source.py [hunk 2]
```diff
@@ -315,6 +318,12 @@ def build_service_principal_map(self) -> None:
                 "service-principals", f"Unable to fetch service principals: {e}"
             )
 
+    def build_groups_map(self) -> None:
+        try:
+            self.groups += self.unity_catalog_api_proxy.groups()
+        except Exception as e:
+            self.report.report_warning("groups", f"Unable to fetch groups: {e}")
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/unity/source.py:325]: Nice ! 

As a result of this graceful handling, I don't see any harm in getting this PR in.

File: metadata-ingestion/src/datahub/ingestion/source/unity/proxy.py [hunk 3]
```diff
@@ -187,6 +187,16 @@ def service_principals(self) -> Iterable[ServicePrincipal]:
...
                 yield optional_sp
 
+    def groups(self):
+        """
+        fetch the list of the groups belongs to the workspace, using the workspace client
+        create the list of group's display name, iterating through the list of groups fetched by the workspace client
+        """
+        group_list: List[str] = []
+        for group in self._workspace_client.groups.list():
+            group_list.append(str(group.display_name))
```
[the remark this answers is not in the document]
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/unity/proxy.py:197]: Let's remove it if its not.

[the remark this answers is not in the document]
Comment (dushayntAW) [reply metadata-ingestion/src/datahub/ingestion/source/unity/proxy.py:197]: removed

[the remark this answers is not in the document]
Comment (dushayntAW) [reply metadata-ingestion/src/datahub/ingestion/source/unity/proxy.py]: removed

**Review by mayurinehate — APPROVED** [review]
LGTM

**Review by anshbansal — APPROVED** [review]

**Pull request merged**